src/projections/igh_o.cpp

```diff
@@ -0,0 +1,235 @@
+#define PJ_LIB__
+
+#include <errno.h>
+#include <math.h>
+
+#include "proj.h"
+#include "proj_internal.h"
+
+PROJ_HEAD(igh_o, "Interrupted Goode Homolosine Oceanic View") "\n\tPCyl, Sph";
+
+C_NAMESPACE PJ *pj_sinu(PJ *), *pj_moll(PJ *);
+
+/* 40d 44' 11.8" [degrees] */
+/*
+static const double d4044118 = (40 + 44/60. + 11.8/3600.) * DEG_TO_RAD;
+has been replaced by this define, to eliminate portability issue:
+Initializer element not computable at load time
+*/
+#define d4044118 ((40 + 44/60. + 11.8/3600.) * DEG_TO_RAD)
+
+static const double d10  =  10 * DEG_TO_RAD;
+static const double d40  =  40 * DEG_TO_RAD;
+static const double d50  =  50 * DEG_TO_RAD;
+static const double d60  =  60 * DEG_TO_RAD;
+static const double d70  =  70 * DEG_TO_RAD;
+static const double d75  =  75 * DEG_TO_RAD;
+static const double d90  =  90 * DEG_TO_RAD;
+static const double d110 = 110 * DEG_TO_RAD;
+static const double d120 = 120 * DEG_TO_RAD;
+static const double d130 = 130 * DEG_TO_RAD;
+static const double d150 = 150 * DEG_TO_RAD;
+static const double d180 = 180 * DEG_TO_RAD;
+
+static const double EPSLN = 1.e-10; /* allow a little 'slack' on zone edge positions */
+
+namespace { // anonymous namespace
+struct pj_opaque {
+    struct PJconsts* pj[12]; \
+    double dy0;
+};
+} // anonymous namespace
+
+
+static PJ_XY igh_o_s_forward (PJ_LP lp, PJ *P) {           /* Spheroidal, forward */
+    PJ_XY xy;
+    struct pj_opaque *Q = static_cast<struct pj_opaque*>(P->opaque);
+    int z;
+
+    if (lp.phi >=  d4044118) {
+           if (lp.lam <=  -d110) z =  1;
+      else if (lp.lam >=    d50) z =  3;
+      else z = 2;
+    }
+    else if (lp.phi >=  0) {
+           if (lp.lam <=  -d110) z =  4;
+      else if (lp.lam >=    d50) z =  6;
+      else z = 5;
+    }
+    else if (lp.phi >= -d4044118) {
+           if (lp.lam <=  -d75) z =  7;
+      else if (lp.lam >=   d90) z =  9;
+      else z = 8;
+    }
+    else {
+           if (lp.lam <=  -d75) z =  10;
+      else if (lp.lam >=   d90) z =  12;
+      else z = 11;
+    }
```

Comment:
What are these magic numbers? Can they be represented as an enum? That would be more descriptive

Reply:
The new projection parallels as closely as possible the existing `igh.cpp` projection. The whole projection is a composite of 12 sub-projections.  The integers are id numbers that correspond to the ascii table at the bottom of the file

Reply:
I don't care much for consistency with the other projection, that's no good either. Please add comments that explain what's going on here. Or better yet, improve the code :-)

Reply:
Fair point.  Documentation added to explain the numbering scheme.